environment/src/main/java/com/sequenceiq/environment/environment/flow/deletion/handler/ResourceEncryptionDeleteHandler.java

```diff
@@ -57,10 +92,63 @@ public void accept(Event<EnvironmentDeletionDto> environmentDtoEvent) {
         }
     }
 
-    private void deleteEncryptionResources(Environment environment) {
-        /*
-        TODO: Delete the Disk Encryption Set if it exists.
-        */
+    private void deleteEncryptionResources(EnvironmentDto environmentDto, Environment environment, String diskEncryptionSetUrl) {
+        if (environmentEncryptionService.deleteEncryptionSet(diskEncryptionSetUrl)) {
+            LOGGER.info("Successfully deleted diskEncryptionSet");
+        } else {
+            LOGGER.error("Failed to delete diskEncryptionSet");
+        }
+        if (!isSingleResourceGroup(environmentDto)) {
+            String resourceGroupName = getResourceGroupNameFromDES(diskEncryptionSetUrl);
+            if (StringUtils.isNotEmpty(resourceGroupName)) {
+                deleteAzureResourceGroup(resourceGroupName, environment);
+            } else {
+                LOGGER.error("Failed to parse Encryption parameters resource group");
+            }
+        }
+    }
+
+    private String getResourceGroupNameFromDES(String diskEncryptionSetUrl) {
+        String regexString = Pattern.quote("resourceGroups/") + "(.*?)" + Pattern.quote("/providers");
+        Pattern pattern = Pattern.compile(regexString);
+        Matcher matcher = pattern.matcher(diskEncryptionSetUrl);
+
+        if (matcher.find()) {
+            return matcher.group(1);
+        } else {
+            return "";
+        }
```

Comment:
Perhaps better to throw something like an `IllegalStateException` or `IllegalArgumentException`. Not having a match means that that the Resource ID is malformed, the RG part is mandatory.